### PR TITLE
Rmrk docs updates

### PR DESCRIPTION
**Pull request by shibatales**

### CODE REVIEW

File: docs/nfts.md [hunk 1]
```diff
@@ -2,4 +2,79 @@
 title: NFTs
 ---
 
-Docs coming soon.
+### React function component example
+
+The following is a component that implements various tool examples using a previous json dump named `ConsolidatedRemarks` (with the working code equivalent commented out). In a nutshell, the component logs the winners of my contest [bishoujo.store](http://bishoujo.store) with a counter of how many times that individual emoted on the nft.
```
Comment (Swader) [inline docs/nfts.md:7]: You can't refer to personal pronouns in docs. "My" has no meaning here. Remove personal angle please. 

Also, due to the evolving nature of external projects, I'd prefer if we didn't link to them from docs and risk making the content outdated asap. If you're documenting an example, if should be atomic, standalone, encapsulated, not depend on much else.

File: docs/nfts.md [hunk 2]
```diff
@@ -2,4 +2,79 @@
...
+
+```
+// const RMRK_PREFIXES = ['0x726d726b', '0x524d524b'];
+const collectionId = '0488b1a94fd8d61738-BISHOUJO';
+export const NftCollection: React.FC = () => {
+  React.useEffect(() => {
+    // const provider = new WsProvider('wss://node.rmrk.app');
+    const fetchAndConsolidate = async () => {
+      try {
+        // const api = await ApiPromise.create({ provider });
```
Comment (Swader) [inline docs/nfts.md:17]: Commented parts of code are just confusing. Focus on the logic you're demonstrating only. Explain why, what it does, and what the point of this example is (what the reader learns).

File: docs/nfts.md [hunk 3]
```diff
@@ -2,4 +2,79 @@
...
+      }
+    };
+
+    fetchAndConsolidate();
+  }, []);
+  return <div className="collection" />;
+};
+```
+
+### Methods and interfaces used
```
Comment (Swader) [inline docs/nfts.md:74]: Does this mean anything to the new reader?

File: docs/tools-cli.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+---
+title: CLI Helpers
+---
+
+If you end up working with large data sets such as the latest dump from Singular, we recommend using the CLI tools to generate this json data for you, in case you experience a large CPU data spike when testing out your frontend. You can read the original CLI documentation below or in the [rmrk-tools github repo](https://github.com/rmrk-team/rmrk-tools).
```
Comment (Swader) [inline docs/tools-cli.md:5]: Link to latest dump.

File: docs/tools-cli.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import shell from "shelljs";
+
+shell.exec(
+  'rmrk-tools-fetch --ws wss://node.rmrk.app --prefixes=0x726d726b,0x524d524b --append=dumps/latest.json',
+);
+```
+
+### Additional notes
+
+- If you get a build error when you try using the cli helper tools, you need to make sure you install all of your dependencies (for me, it required that I install the dependencies `@polkadot/api` and `arg`).
```
Comment (Swader) [inline docs/tools-cli.md:34]: If we need these and they are not included, this is something to be flagged on the tools repo and fixed.

File: docs/tools-cli.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+> "arg is for local dev cli tools." --Ilia
+
+- For those working with latest dump data: if you run the cli:`fetch` method, and you are still trying to connect to the 127.0.0.1:9944 default endpoint, you need to run fetch with the optional parameter `--ws wss://node.rmrk.app` or whatever you decided on your local node endpoint.
```
Comment (Swader) [inline docs/tools-cli.md:38]: Why?
If this is a tools bug / gaffe, it should be documented there and fixed. We don't want to document workarounds for bugs, not when we are the authors of the bugged tools.

File: docs/tools-cli.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      {
+        call: "system.remark",
+        value: "0x13371338",
+      },
+    ],
+  },
+];
+```
+
+Optional parameters for `fetch`:
```
Comment (Swader) [inline docs/tools-cli.md:81]: You're missing escapes here:

\- --ws URL

vs

\- `--ws URL`


Also not sure this is really necessary - we'll keep the README of the tools up to date. By copying it here, you risk going out of date in two places.

File: docs/tools-docs.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+## In a nutshell, RMRK tools:
+
+- GET complete state of all remarks right now
+- RETRIEVE remarks from extrinsic block call
+- VALIDATE remarks
+- CREATE new remark strings ready to be sent as transactions
+
+This walkthrough of our Tools is based on the 1.0 spec. We will be releasing v2.0 docs whenever RMRK2.0 hits the shelves! For the best developer experience, we recommend developing with the most current **Mac operating system**.
```
Comment (Swader) [inline docs/tools-docs.md:12]: > we recommend developing with the most current **Mac operating system**.

We absolutely do not, lol :D Remove please. Recommend a *nix environment, that's it.

File: docs/tools-docs.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+### Warning: Not a stroll in the park
```
Comment (Swader) [inline docs/tools-docs.md:14]: Unnecessary, why scare people away?

File: docs/tools-docs.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+Getting RMRK Tools to run properly in local development is currently not a plug-and-play experience. You will need to be familiar with creating your first Substrate node using the Rust programming language; take 2 hours to get yourself jumpstarted [here](https://substrate.dev/tutorials).
```
Comment (Swader) [inline docs/tools-docs.md:16]: You can just download the Polkadot or Kusama binary and run it if on Ubuntu. The tutorial to do a compilation, though, is more like 10 minutes + 40 min compilation time.

File: docs/tools-docs.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+### [Install Cargo and Rust](https://doc.rust-lang.org/stable/cargo/getting-started/installation.html)
+
+### Install Rust Nightly
+
+`rustup toolchain install nightly`
+
+### nvm/node
+
+Make sure to update to recent version of nvm/node 14.0+; if you're on Windows, nvm-windows has an installer (but again, I would like to reiterate that this tutorial is for Mac development only).
```
Comment (Swader) [inline docs/tools-docs.md:26]: Again personal angle, please make more objective. Do not recommend any OS beyond using a *nix ecosystem: Ubuntu on Linux, Ubuntu on WSL if on Windows, OS X on Mac. That's all people need to know.

File: docs/tools-docs.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+### polkadot.js.org
+
+Make sure that in polkadot.js.org that you switch to the Development network using the custom endpoint: `wss://node.rmrk.app` or whatever local endpoint you decided on (press the Switch button at the top to finalize the selection!)
```
Comment (Swader) [inline docs/tools-docs.md:30]: Don't recommend `node.rmrk.app` when mentioning dev environments, as people might take it to mean that's a dev endpoint.

File: docs/tools-docs.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+### pallet-utility
+
+> "Keep in mind that if it's a node template you'll need to install the utility pallet into it; RMRK requires it. Or compile polkadot--that one has pallet-utility built in." --Bruno
+
+You need to install `pallet-utility` in Cargo.toml under the runtime directory.
```
Comment (Swader) [inline docs/tools-docs.md:36]: Skip this whole section, if they download and/or compile substrate-node, Polkadot, or Kusama, it's all taken care of.

File: docs/tools-docs.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+### Latest Singular dump
+
+If you want to test your UI with the latest data dump from Singular: [💩💩💩](https://gateway.pinata.cloud/ipns/latestdump.rmrk.app)
```
Comment (Swader) [inline docs/tools-docs.md:40]: Let's be professional

```suggestion
If you want to test your UI with the [latest data dump from Singular](https://gateway.pinata.cloud/ipns/latestdump.rmrk.app)
```

**Review by Swader — CHANGES_REQUESTED** [review]
Good start but needs lots more love and work.

**Review by rdovgan — APPROVED** [review]

**Comment by Swader** [comment]
This has kind of expired and the docs significantly changed since

**Pull request closed without merging**